processor/src/operations/io_ops.rs

```diff
@@ -249,6 +249,14 @@ where
         Ok(())
     }
 
+    /// Forwards the emitted event id to the host.
+    pub(super) fn op_emit(&mut self, event_id: u32) -> Result<(), ExecutionError> {
+        self.stack.copy_state(0);
+        self.host.borrow_mut().on_event(self, event_id)?;
+
+        Ok(())
+    }
```

Comment:
We should probably move this into `operations/sys.rs` module.

Reply:
done